gateway-shell/src/main/java/org/apache/knox/gateway/shell/hdfs/Ls.java

```diff
@@ -41,6 +42,11 @@ public Request dir( String dir ) {
       return this;
     }
 
+    public Request knoxToken(String knoxToken) {
```

Comment:
Why not add Knox token support to KnoxSession?

Reply:
IMO, the token handling should happen on request level. But you are right, this should not be limited to the `ls` command -> moving it to the abstract parent.

Reply:
Fixed.